code-challenges.test.js

```diff
@@ -21,8 +21,53 @@ const secretCodeWord2 = "Gobbledygook"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+describe("code", ()=>{
+    it("takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", ()=>{
+        expect(code(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(code(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(code(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+
 // b) Create the function that makes the test pass.
 
+//Pseudo Code
+//Create a function named code
+// function takes in a string
+//create a variable to put new string into
+//create logic that pushes numbers instead of strings when the number is equal to A, E, I, or O.
+//use a for loop to iterate over the string to find if it has those letters
+//push it onto new string along with whatever else was in the word
+
+const code = (string) => {
+    let newString = ""
+    for(let i = 0; i < string.length; i++) {
+        if(string[i] === 'a') {
+            newString[i]("4")
+        } 
+        else if(string[i] === 'e') {
+            newString[i]("3") 
+            }
+        else if(string[i] === 'e') {
+            newString[i]("3")
+    }
+        else if(string[i] === 'i') {
+            newString[i]("1")
+}
+        else if(string[i] === 'i') {
+            newString[i]("1")
+        }
+    }
+    return newString
```

Comment:
This is 90% good. Just missing some edge cases of capital letters and you are all set.